### PR TITLE
decorate pr

### PR DESCRIPTION
**Pull request by lusol**
Adding functions to decorate new pr field in events

Related to https://github.com/screwdriver-cd/screwdriver/issues/822

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -224,6 +224,25 @@ class ScmBase {
...
+     * @return {Promise}
+     */
+    decoratePR(config) {
+        return validate(config, dataSchema.plugins.scm.decoratePR)
+            .then(validPR => this._decoratePR(validPR))
+            .then(decoratePR => validate(decoratePR, dataSchema.core.scm.pr));
+    }
+
+    _decoratePR() {
+        return Promise.reject(new Error('Not implemented'));
```
Comment (Zhongtang) [inline index.js]: If we reject it here, we need to make sure `scm-bitbucket` also implements this method with a dummy return value.

Comment (lusol) [reply index.js]: yeah, I'm implementing it in all the other scm repos

File: index.js [hunk 2]
```diff
@@ -224,6 +224,25 @@ class ScmBase {
         return Promise.reject(new Error('Not implemented'));
     }
 
+     /**
+     * Decorate the author for the specific source control
+     * @method decorateAuthor
```
Comment (tkyi) [inline index.js]: decoratePR*

File: index.js [hunk 3]
```diff
@@ -224,6 +224,25 @@ class ScmBase {
         return Promise.reject(new Error('Not implemented'));
     }
 
+     /**
+     * Decorate the author for the specific source control
```
Comment (tkyi) [inline index.js]: Decorate the PR*

**Comment by tkyi** [comment]
Can you link this PR to the issue in the description?

File: index.js [hunk 4]
```diff
@@ -358,7 +358,8 @@ class ScmBase {
             .then(pr => validate(pr, Joi.object().keys({
                 name: Joi.reach(dataSchema.models.job.base, 'name').required(),
                 sha: Joi.reach(dataSchema.models.build.base, 'sha').required(),
-                ref: Joi.string().required()
+                ref: Joi.string().required(),
+                url: Joi.string().url()
```
Comment (tkyi) [inline index.js]: can probably do
```
url: Joi.reach(dataSchema.core.scm.pr, 'url')
```

File: index.js [hunk 5]
```diff
@@ -358,7 +358,8 @@ class ScmBase {
             .then(pr => validate(pr, Joi.object().keys({
                 name: Joi.reach(dataSchema.models.job.base, 'name').required(),
                 sha: Joi.reach(dataSchema.models.build.base, 'sha').required(),
-                ref: Joi.string().required()
+                ref: Joi.string().required(),
+                url: Joi.string().uri()
```
Comment (tkyi) [inline index.js]: Can do `url: Joi.reach(dataSchema.core.scm.pr, 'url')`

File: README.md [hunk 6]
```diff
@@ -347,10 +347,10 @@ The parameters required are:
 | config.scmContext | String | No | The name of scm context |
 
 #### Expected Outcome
-The object consists of PR name, sha and ref for the pipeline.
+The object consists of PR name, sha, ref, and url for the pipeline.
 
 #### Expected Promise Response
-1. Resolve with the object consists of PR name, sha and ref
+1. Resolve with the object consists of PR name, sha, and ref
```
Comment (minz1027) [inline README.md]: why no url here? can put `url(optional)`

**Pull request merged**